apps/src/lib/node/ledger/protocol/transactions/ethereum_events/eth_msg_updates.rs

```diff
@@ -0,0 +1,162 @@
+use itertools::Itertools;
+use namada::types::ethereum_events::vote_extensions::MultiSignedEthEvent;
+use namada::types::ethereum_events::EthMsgUpdate;
+
+pub(crate) fn from_multisigneds(
+    multisigneds: Vec<MultiSignedEthEvent>,
+) -> Vec<EthMsgUpdate> {
+    multisigneds.into_iter().map(from_multisigned).collect()
+}
+
+// Derive the [`EthMsgUpdate`] for some given [`MultiSignedEthEvent`]. This
+// function is deterministic.
+pub(crate) fn from_multisigned(
+    multisigned: MultiSignedEthEvent,
+) -> EthMsgUpdate {
+    let body = multisigned.event;
+
+    let seen_by = multisigned
+        .signers
+        .into_iter()
+        .sorted_by(|a, b| a.cmp(b))
+        .collect();
+
+    EthMsgUpdate { body, seen_by }
+}
```

Comment:
I'm not sure now whether these fns need to be deterministic. When acting on Ethereum events we need to do so in a deterministic order, but for updating of `/eth_msgs` storage I'm not sure if this is strictly necessary. eg when adding voting powers to an EthMsgs, or updating the `seen_by` field. All nodes should arrive at the same result regardless, and the `seen_by` field should always be saved in some deterministic sorted order, but the intermediary calculations/additions can be in any order.